### ISSUE TITLE
A module subroutine is wrongly taken as function.

### ISSUE THREAD

**Issue by h-murai** [omni-compiler/omni-compiler#460]
The code causes an error:

`"hoge.f90:mmm", line 10: 'sub2' is a function, not a subroutine`

```fortran
module mmm

  implicit none
  public :: sub1, sub2

contains

  subroutine sub1
    call sub2
    call sub2
  end subroutine sub1

  subroutine sub2
  end subroutine sub2

end module mmm
```

**Comment by clementval** [comment]
@h-murai This might solve the problem of #445, I'm running my test right now. If so, I'll just close the #445 otherwise I'll provide a test case.

**Comment by clementval** [comment]
@h-murai @shingo-s I confirm that this fix solve the problem I was facing in #445
